### PR TITLE
Us058 update respondent email

### PR DESCRIPTION
**Pull request by rhiGriff**
Adding a workaround for updating a respondents email for release 17

### CODE REVIEW

File: response_operations_ui/controllers/edit_contact_details_controller.py [hunk 1]
```diff
@@ -14,12 +14,22 @@ def edit_contact_details(edit_details_data, respondent_id):
...
-        return False
-
-    logger.debug('Successfully changed contact details', respondent_id=respondent_id)
-    return True
+    if response.status_code == 200:
+        logger.debug('Respondent details updated', respondent_id=respondent_id, status_code=response.status_code)
+        return True, ''
+
+    if response.status_code == 409:
+        logger.debug('Email address already exists', respondent_id=respondent_id, status_code=response.status_code)
```
Comment (LJBabbage) [inline response_operations_ui/controllers/edit_contact_details_controller.py]: This information could be usefully so I would change it to info, rather then debug

File: response_operations_ui/controllers/edit_contact_details_controller.py [hunk 2]
```diff
@@ -14,12 +14,22 @@ def edit_contact_details(edit_details_data, respondent_id):
...
-        logger.debug('Error changing contact details', respondent_id=respondent_id)
-        return False
-
-    logger.debug('Successfully changed contact details', respondent_id=respondent_id)
-    return True
+    if response.status_code == 200:
+        logger.debug('Respondent details updated', respondent_id=respondent_id, status_code=response.status_code)
+        return True, ''
+
+    if response.status_code == 409:
```
Comment (LJBabbage) [inline response_operations_ui/controllers/edit_contact_details_controller.py]: I dug into backstage and then party, but couldn't see where a 409 was generated, change_respondent_details in party is slightly odd

File: response_operations_ui/forms.py [hunk 3]
```diff
@@ -45,6 +45,7 @@ class EditContactDetailsForm(FlaskForm):
     last_name = StringField('last_name')
     email = StringField('emailAddress')
     telephone = StringField('telephone')
+    hidden_email = HiddenField('hidden_email')
```
Comment (LJBabbage) [inline response_operations_ui/forms.py]: Whats does hidden email mean?

File: response_operations_ui/controllers/edit_contact_details_controller.py [hunk 4]
```diff
@@ -14,12 +14,22 @@ def edit_contact_details(edit_details_data, respondent_id):
...
+        logger.debug('Email address already exists', respondent_id=respondent_id, status_code=response.status_code)
+        return False, 'bad-email'
```
Comment (LJBabbage) [inline response_operations_ui/controllers/edit_contact_details_controller.py]: I would prefer not to return to 2 things from this function just to cater for one error_type, feels like these should be raising an exception

File: response_operations_ui/controllers/edit_contact_details_controller.py (hunk 1 above)
Comment (LJBabbage) [inline response_operations_ui/controllers/edit_contact_details_controller.py]: If there there is a 409, could it be a conflict other then the email?

File: response_operations_ui/templates/edit-contact-details.html [hunk 5]
```diff
@@ -89,6 +97,7 @@ <h2 id='error-saving-message' class="venus">Error saving changes</h2>
                       <span class="label__description pluto">ONS will send a verification email to a new address</span>
                     </label>
                   {{ form.email(class_="input input--text", id='email', required=required, maxlength=254, type="email") }}
+                    <input type="hidden" name="hidden_email" value={{ respondent_details.emailAddress }}>
```
Comment (LJBabbage) [inline response_operations_ui/templates/edit-contact-details.html]: Could someone hack this?

File: response_operations_ui/views/reporting_units.py [hunk 6]
```diff
@@ -87,19 +87,47 @@ def edit_contact_details(ru_ref, respondent_id):
...
+            return render_template('edit-contact-details.html', ru_ref=ru_ref, form=form, email_error=True,
+                                   respondent_details=respondent_details)
+        else:
+            return render_template('edit-contact-details.html', ru_ref=ru_ref, form=form, error=True,
+                                   respondent_details=respondent_details)
+
+    message = None
+    details_changed = False
+    email_changed = False
+
```
Comment (LJBabbage) [inline response_operations_ui/views/reporting_units.py]: These 'ifs' feels like they should be in the controller, and the controller should just return what has changed

**Comment by LJBabbage** [comment]
@rhiGriff I have pushed the bits I was talking to you about. Hopefully its easiest enough to follow

File: response_operations_ui/forms.py (hunk 3 above)
Comment (tej99) [inline response_operations_ui/forms.py]: If this is just the user's present email address it doesn't look like you need this field. Could you not just use the email address you get back when making the 'get_contact_details(respondent_id)' call on line 38 in contact_details_controllers.py?

File: response_operations_ui/controllers/contact_details_controller.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+def update_contact_details(ru_ref, respondent_id, form):
+
+    new_contact_details = {
+        "first_name": form.get('first_name'),
+        "last_name": form.get('last_name'),
+        "email_address": form.get('hidden_email'),
+        "new_email_address": form.get('email'),
+        "telephone": form.get('telephone'),
+        "respondent_id": respondent_id}
```
Comment (andrewmil) [inline response_operations_ui/controllers/contact_details_controller.py]: Is respondent_id required here?

**Comment by rhiGriff** [comment]
[US058: INT Change Respondent Email Address [XL]](https://trello.com/c/KQUOeZBV/46-us058-int-change-respondent-email-address-xl)

**Pull request merged**